### PR TITLE
Backend for friends routes

### PR DESCRIPTION
**Pull request by SsaaaadD2**
What it does
- Added backend routes to add or remove friends
- Routes to make, cancel and reject friend requests
- Suggested friends: returns 20 random users from the database
- Route to return all the friends for a user
- Routes to see all requests made by a user and all requests received by a user

What to look for
server/friends/friend.js

### CODE REVIEW

**Review by aeciorc — CHANGES_REQUESTED** [review]
This looks good, but to keep the API cleaner, let's use the RESTful naming convention for routes. https://restfulapi.net/resource-naming/
For example, /friendRequests POST to create a friend request, DELETE to remove it 
/friends POST to accept an invitation and add a friend, DELETE to remove a friend

Doesn't have to be exactly like this, but something along these lines

File: server/routes/auth/register.js [hunk 1]
```diff
@@ -83,8 +82,9 @@ router.post(
       //   }
       // );
 
-      jwtSign(payload, process.env.JWT_SECRET, 1.577e7, res, userObject);
```
Comment (TheAllen) [inline server/routes/auth/register.js:86]: The jwtSign function with the time parameter is the correct jwtSign function

File: server/routes/auth/signin.js [hunk 2]
```diff
@@ -68,9 +68,10 @@ router.post(
       //     res.json({ sucess: true, token, userObject });
       //   }
       // );
-
-      jwtSign(payload, process.env.JWT_SECRET, 1.577e7, res, userObject);
```
Comment (TheAllen) [inline server/routes/auth/signin.js:72]: Same here

**Review by TheAllen — COMMENTED** [review]
The routes look good to me. I think Aecio mention the better naming conventions for the routes. The jwtSign function that include the expireTime should be the right jwtSign function.

File: server/routes/auth/register.js (hunk 1 above)
Comment (SsaaaadD2) [reply server/routes/auth/register.js:86]: Okay, should be fixed now

**Comment by SsaaaadD2** [comment]
> This looks good, but to keep the API cleaner, let's use the RESTful naming convention for routes. https://restfulapi.net/resource-naming/
> For example, /friendRequests POST to create a friend request, DELETE to remove it
> /friends POST to accept an invitation and add a friend, DELETE to remove a friend
> 
> Doesn't have to be exactly like this, but something along these lines

Okay, I've updated the route names

File: server/routes/friends/friend.js [hunk 3]
```diff
@@ -0,0 +1,326 @@
+const express = require("express");
+const router = express.Router();
+const User = require("../../models/User");
+
+const authentication = require("../../middleware/authentication");
+
+//@route            POST /api/friend/:id
+//@desc             Make this user a friend of the current user
+//@access           Private
+router.post("/friend/:id", [authentication], async function (req, res) {
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.post("/friends, [authentication], async function (req, res) { 
```
With the friend_id coming from the body

File: server/routes/friends/friend.js [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+      return res.status(400).json({msg: "User does not exist."});
+    }
+    res.status(500).send("Server error");
+  }
+});
+
+//@route            DELETE /api/remove-friend/:id
+//@desc             Remove this friend or delete the sent friend request
+//@access           Private
+router.delete("/friend/:id", [authentication], async function (req, res) {
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.delete("/friends/:id", [authentication], async function (req, res) {
```

File: server/routes/friends/friend.js [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+      return res.status(400).json({msg: "User does not exist."});
+    }
+    res.status(500).send("Server error");
+  }
+});
+
+//@route            POST /api/friend-request/:id
+//@desc             Make this user a friend of the current user
+//@access           Private
+router.post("/friend-request/:id", [authentication], async function (req, res) {
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.post("/outgoing-requests", [authentication], async function (req, res) {
```
with the friend_id coming from the body

File: server/routes/friends/friend.js [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+      return res.status(400).json({msg: "User does not exist."});
+    }
+    res.status(500).send("Server error");
+  }
+});
+
+//@route            DELETE /api/cancel-friend-request/:id
+//@desc             Remove this friend or delete the sent friend request
+//@access           Private
+router.delete("/friend-request/:id", [authentication], async function (
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.delete("/outgoing-requests/:id", [authentication], async function (
```

File: server/routes/friends/friend.js [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+      return res.status(400).json({msg: "User does not exist"});
+    }
+    res.status(500).send("Server error");
+  }
+});
+
+//@route            DELETE /api/reject-friend-request/:id
+//@desc             Reject friend request made by a user
+//@access           Private
+router.delete("/reject-friend-request/:id", [authentication], async function (
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.delete("/received-requests/:id", [authentication], async function (
```

File: server/routes/friends/friend.js [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+  } catch (error) {
+    console.log(error.message);
+    res.status(500).send("Server error");
+  }
+});
+
+//@route                GET /api/friend-requests-sent
+//@desc                 Get all the friend requests sent by a user
+//@access               Private
+router.get("/friend-requests-sent", [authentication], async function (
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.get("/outgoing-requests", [authentication], async function (
```

File: server/routes/friends/friend.js [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+  } catch (error) {
+    console.log(error.message);
+    res.status(500).send("Server error");
+  }
+});
+
+//@route                GET /api/friend-requests-received
+//@desc                 Get all the friend requests made to a user
+//@access               Private
+router.get("/friend-requests-received", [authentication], async function (
```
Comment (aeciorc) [inline server/routes/friends/friend.js]:
```suggestion
router.get("/received-requests", [authentication], async function (
```

**Review by aeciorc — COMMENTED** [review]
Left some suggestions on how to rename the routes to fit the convention and the name of your entities

**Review by aeciorc — APPROVED** [review]

**Pull request merged**